### PR TITLE
[AIRFLOW-3939] Add Google Cloud Translate operator

### PR DESCRIPTION
**Pull request by potiuk**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. 
  - https://issues.apache.org/jira/browse/AIRFLOW-3939

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:
Add Google Cloud Translate operator

### Tests

- [x] My PR adds the following unit tests
* TestCloudTranslateHook
* CloudTranslateTest
* CloudTranslateExampleDagsSystemTest

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
  - All the public functions and the classes in the PR contain docstrings that explain what it does

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -180,6 +180,7 @@ def write_version(filename=os.path.join(*['airflow',
     'google-cloud-bigtable==0.31.0',
     'google-cloud-spanner>=1.7.1',
     'google-cloud-vision>=0.35.2',
+    'google-cloud-translate>=1.3.3',
```
Comment (zhongjiajie) [inline setup.py:183]: `google-cloud-translate>=1.3.3` before `google-cloud-vision>=0.35.2` for alphabetical?

File: tests/contrib/hooks/test_gcp_translate_hook.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    from unittest import mock
+except ImportError:
+    try:
+        import mock
+    except ImportError:
+        mock = None
+
+PROJECT_ID_TEST = 'project-id'
+
+
```
Comment (zhongjiajie) [inline tests/contrib/hooks/test_gcp_translate_hook.py:35]: In L31, mock could be `None`, should we use in L35?
```suggestion
@unittest.skipIf(mock is None, 'mock package not present')
```
So as `test_gcp_translate_operator.py`

File: airflow/contrib/operators/gcp_translate_operator.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+      for translation. Sets the ``cid`` parameter
+      in the query.
+
+    :type model: str or None
+    :param model: (Optional) The model used to translate the text, such
+      as ``'base'`` or ``'nmt'``.
+
+    """
+
+    # [START translate_template_fields]
```
Comment (zhongjiajie) [inline airflow/contrib/operators/gcp_translate_operator.py:74]: Could we remove this line and L81?

File: airflow/contrib/hooks/gcp_translate_hook.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+                  * ``input``: The corresponding input value.
+                  * ``model``: The model used to translate the text.
+
+                  If only a single value is passed, then only a single
+                  dictionary will be returned.
+        :raises: :class:`~exceptions.ValueError` if the number of
+                 values and translations differ.
+        """
+        client = self.get_conn()
+
```
Comment (zhongjiajie) [inline airflow/contrib/hooks/gcp_translate_hook.py:90]: Should we add some log for debug easy?

**Review by zhongjiajie — COMMENTED** [review]
Please find my two cents.

File: setup.py (hunk 1 above)
Comment (potiuk) [reply setup.py:183]: Good catch

File: tests/contrib/hooks/test_gcp_translate_hook.py (hunk 2 above)
Comment (potiuk) [reply tests/contrib/hooks/test_gcp_translate_hook.py:35]: This pattern is sprinkled all-over the Airflow code and copy&pasted everywhere. It's rather unlikely for mock to be None - in different python versions you simply expect it in either unittests or directly imported. I think this problem will be gone fairly soon when Airflow 2.0 drops python 2.7 compatibility, and until then I'd rather leave it as is.

File: airflow/contrib/operators/gcp_translate_operator.py (hunk 3 above)
Comment (potiuk) [reply airflow/contrib/operators/gcp_translate_operator.py:74]: We are using those to generate user Sphinx-generated documentation automatically and include pieces of actual working code in the documentation. We are showing which fields are templated to the user of the operator. See https://github.com/apache/airflow/pull/4755/files#diff-40e5e831dc73c82a2aba583e5f89f8a8R2372  

It's really useful as the documentation is rather up-to-date this way.

File: airflow/contrib/hooks/gcp_translate_hook.py (hunk 4 above)
Comment (potiuk) [reply airflow/contrib/hooks/gcp_translate_hook.py:90]: We already have a debug log in the Operator. https://github.com/apache/airflow/pull/4755/files#diff-250e99f89f67f9e59efc842ed614fb59R116 which I used to see the results. And it makes sense to put it there usually - especially that sometimes several different hooks are used in single "execute" method of the operator and we usually add some logging there (as needed).

File: setup.py (hunk 1 above)
Comment (potiuk) [reply setup.py:183]: Fixed.

File: airflow/contrib/hooks/gcp_translate_hook.py (hunk 4 above)
Comment (zhongjiajie) [reply airflow/contrib/hooks/gcp_translate_hook.py:90]: Ok, I get this.

File: tests/contrib/hooks/test_gcp_translate_hook.py (hunk 2 above)
Comment (zhongjiajie) [reply tests/contrib/hooks/test_gcp_translate_hook.py:35]: got

File: airflow/contrib/operators/gcp_translate_operator.py (hunk 3 above)
Comment (zhongjiajie) [reply airflow/contrib/operators/gcp_translate_operator.py:74]: oh... a bad review :joy:

**Comment by zhongjiajie** [comment]
@potiuk The `docs` are change, maybe you should rebase on master.

**Comment by mik-laj** [comment]
@zhongjiajie This conflict is caused by my change - https://github.com/apache/airflow/pull/4814

Now it is not advisable to update this PR for two reasons. First, further changes may occur, because we're working on the documentation structure all the time (ex. https://github.com/apache/airflow/pull/4894). They only expand the scope of documentation. If the current documentation is sufficient, future versions also. Secondly, this change can not be accepted, because commiter is on vacation. There is only one person who can accept these changes. If he comes back from holidays, we will pass him other changes that were made first.

I hope to understand and accept our way of working.

**Comment by zhongjiajie** [comment]
@mik-laj Get it

**Comment by potiuk** [comment]
@mik-laj  @zhongjiajie  -> actually I rebased it now and solved the conflict/moved the translate operator to separate translate.rst following #4814 convention (I love the way documentation got improved recently - mostly by @mik-laj  ;) ). 
It's quite OK to do such rebase now it takes few minutes and this way next rebase even if we change structure a bit will be easier.

**Comment by zhongjiajie** [comment]
@potiuk Nice! but CI failed random, maybe you should force push make it green

**Comment by potiuk** [comment]
Hey @kaxil -> I know you must have been busy after holidays (I hope you rested :)),  but just wanted to drag your attention to this small PR ;).

File: airflow/contrib/example_dags/example_gcp_translate.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+from airflow.contrib.operators.gcp_translate_operator import CloudTranslateTextOperator
+from airflow.operators.bash_operator import BashOperator
+
+default_args = {'start_date': airflow.utils.dates.days_ago(1)}
+
+with models.DAG(
+    'example_gcp_translate', default_args=default_args, schedule_interval=None  # Override to match your needs
+) as dag:
+    # [START howto_operator_translate_text]
+    product_set_create = CloudTranslateTextOperator(
```
Comment (mik-laj) [inline airflow/contrib/example_dags/example_gcp_translate.py]:
```suggestion
    translate = CloudTranslateTextOperator(
```

File: airflow/contrib/example_dags/example_gcp_translate.py [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        task_id='translate',
+        values=['zażółć gęślą jaźń'],
+        target_language='en',
+        format_='text',
+        source_language=None,
+        model='base',
+    )
+    # [END howto_operator_translate_text]
+    # [START howto_operator_translate_access]
+    translation_access = BashOperator(
```
Comment (mik-laj) [inline airflow/contrib/example_dags/example_gcp_translate.py]:
```suggestion
    translate_result = BashOperator(
```

File: airflow/contrib/example_dags/example_gcp_translate.py [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+        task_id='access',
```
Comment (mik-laj) [inline airflow/contrib/example_dags/example_gcp_translate.py]:
```suggestion
        task_id='translate_result',
```
I have adopted such conventions in my other examples.

File: airflow/contrib/example_dags/example_gcp_translate.py [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+        bash_command="echo '{{ task_instance.xcom_pull(\"translate\")[0] }}'"
+    )
+    product_set_create >> translation_access
```
Comment (mik-laj) [inline airflow/contrib/example_dags/example_gcp_translate.py]:
```suggestion
    translate >> translate_result
```

**Review by kaxil — APPROVED** [review]
LGTM

**Comment by kaxil** [comment]
@potiuk Thanks and yeah I am still catching up with my office and work and its too busy. :) Should have more time from coming week.

**Pull request merged**